app/controllers/samples_controller.rb

```diff
@@ -533,7 +539,16 @@ def samples_going_public
       params[:userId] ? User.find(params[:userId]) : current_user,
       params[:projectId] ? Project.find(params[:projectId]) : nil
     )
-    render json: samples.to_json(include: [{ project: { only: [:id, :name] } }])
```

Comment:
@tfrcarvalho FYI: `samples.to_json(include: :project)` will make a SQL query for each record in samples. A `samples.includes(:project).each` loop will not.

Reply:
The problem here is not the to_json, but the no inclusion/join of project before as in `samples.joins(:project).to_json(...`

Reply:
got it